src/amp-story-player/amp-story-player-impl.js

```diff
@@ -393,6 +431,144 @@ export class AmpStoryPlayer {
     this.rootEl_.appendChild(button);
   }
 
+  /**
+   * Creates the up next elements, attaches event listeners so the
+   * countdown will begin when a user reaches the end of the story and
+   * be cleared after navigation or the cancel button is triggered.
+   * If the countdown successfully reaches 0, the player will display
+   * the next story.
+   * @private
+   */
+  initializeUpNext_() {
+    if (this.stories_.length === 1) {
+      return;
+    }
+
+    const {container, card, text, title, author} = this.createUpNextElement_();
+
+    this.element_.addEventListener('storyEnd', () => {
```

Comment:
We'll need to send this as a post message from the story, and register that handler in the player.